File: solution/ServerExam/Entity/QuestionSolutionEntity.cs
﻿using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace ServerExam.Entity
{
    public class QuestionSolutionEntity
    {
        [Key]
        [JsonPropertyName("ID")]
        public int ID { get; set; }

        /// <summary>
        /// 试题ID
        /// </summary>
        [JsonPropertyName("QuestionID")]
        public int QuestionID { get; set; }

        /// <summary>
        /// 选项
        /// </summary>
        [JsonPropertyName("Option")]
        public string Option { get; set; }

        /// <summary>
        /// 选项附件
        /// </summary>
        [JsonPropertyName("OptionAttachment")]
        public string OptionAttachment { get; set; }

        /// <summary>
        /// 正确答案
        /// </summary>
        [JsonPropertyName("CorrectAnswer")]
        public string CorrectAnswer { get; set; }

        /// <summary>
        /// 考生答案
        /// </summary>
        [JsonPropertyName("CandidateAnswer")]
        public string CandidateAnswer { get; set; }

        /// <summary>
        /// 得分比例
        /// </summary>
        [JsonPropertyName("ScoreRatio")]
        public decimal ScoreRatio { get; set; }

        /// <summary>
        /// 创建时间
        /// </summary>
        [JsonPropertyName("CreateTime")]
        public int CreateTime { get; set; }

        /// <summary>
        /// 更新时间
        /// </summary>
        [JsonPropertyName("UpdateTime")]
        public int UpdateTime { get; set; }

        public QuestionSolutionEntity()
        {
            this.ID = 0;
            this.QuestionID = 0;
            this.Option = "";
            this.OptionAttachment = "";
            this.CorrectAnswer = "";
            this.CandidateAnswer = "";
            this.ScoreRatio = 0;
            this.CreateTime = 0;
            this.UpdateTime = 0;
        }
    }
}

File: solution/Entity/QuestionSolutionEntity.py
# -*- coding:utf-8 -*-
from Entity.BaseEntity import *


# 试题选项
class QuestionSolutionEntity(BaseEntity, BaseORM):
    __tablename__ = 'QuestionSolution'

    QuestionID: int = Column(INTEGER, comment='试题ID', default=0)
    Option: str = Column(String(128), comment='试题选项', default='none')
    OptionAttachment: str = Column(String(65535), comment='试题附件', default='none')
    CorrectAnswer: int = Column(INTEGER, comment='正确答案 1错误 2正确', default=0)
    CorrectItem: str = Column(String(65535), comment='答案项', default='none')
    ScoreRatio: float = Column(DECIMAL(10, 2), comment='得分比例', default=0.00)
    Position: int = Column(INTEGER, comment='拖拽题/连线题 展示位置 1左 2右', default=0)
    UpdateTime: int = Column(INTEGER, comment='更新时间', default=int(time()))

    def __init__(self):
        super().__init__()